x/ibc/light-clients/solomachine/types/solomachine.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"github.com/tendermint/tendermint/crypto"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+)
+
+var _, _, _, _ codectypes.UnpackInterfacesMessage = &ClientState{}, &ConsensusState{}, &Header{}, &HeaderData{}
+
+// UnpackInterfaces implements the UnpackInterfaceMessages.UnpackInterfaces method
+func (cs *ClientState) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	return cs.ConsensusState.UnpackInterfaces(unpacker)
+}
+
+// UnpackInterfaces implements the UnpackInterfaceMessages.UnpackInterfaces method
+func (cs *ConsensusState) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	var pubKey crypto.PubKey
+	err := unpacker.UnpackAny(cs.PublicKey, &pubKey)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UnpackInterfaces implements the UnpackInterfaceMessages.UnpackInterfaces method
+func (h *Header) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	var pubKey crypto.PubKey
+	err := unpacker.UnpackAny(h.NewPublicKey, &pubKey)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UnpackInterfaces implements the UnpackInterfaceMessages.UnpackInterfaces method
+func (hd *HeaderData) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	var pubKey crypto.PubKey
+	err := unpacker.UnpackAny(hd.NewPubKey, &pubKey)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
let's add a test for unpacking Header and HeaderData

Reply:
Hmm, I'm not sure it will be super useful. It will be a copy-paste of [this test](https://github.com/cosmos/cosmos-sdk/blob/5c0e3b4de5b808765b1ec4793cfdba30401e023e/codec/types/types_test.go#L64) basically, only with `s/testdata.HasAnimal/Header`.

If you think it's useful, I can still add one.

Reply:
The types are different. But if you say that it won't test anything new, then OK.

Reply:
yeah, you're right, it's better to have too many tests than too few.